src/devices/Mpu9250/Mpu9250.cs

```diff
@@ -0,0 +1,983 @@
+        /// Only Accelerator data is availablein this mode
+        /// </summary>
+        /// <param name="magnetometerThreshold"></param>
+        /// <param name="acceleratorLowPower"></param>
+        public void SetWakeOnMotion(uint magnetometerThreshold, AccelerometerLowPowerFrequency acceleratorLowPower)
+        {
+            // Using documentation page 31 of Product Specification to setup
+            _wakeOnMotion = true;
+            if (magnetometerThreshold > 1020)
+                throw new ArgumentException($"{nameof(magnetometerThreshold)} has to be between 0mg and 1020mg");
+            // LSB = 4mg
+            magnetometerThreshold /= 4;
+            // We can't use the magnetometer, only Accelerometer will be measured
+            _ak8963.MeasurementMode = MeasurementMode.PowerDown;
+            // Make sure we start from a clean soft reset
+            PowerOn();
+            //  PWR_MGMT_1 (0x6B) make CYCLE =0, SLEEP = 0  and STANDBY = 0 
+            WriteRegister(Register.PWR_MGMT_1, (byte)ClockSource.Internal20MHz);
+            // PWR_MGMT_2 (0x6C) set DIS_XA, DIS_YA, DIS_ZA = 0 and DIS_XG, DIS_YG, DIS_ZG = 1
+            // Remove the Gyroscope
+            WriteRegister(Register.PWR_MGMT_2, (byte)(DisableModes.DisableGyroscopeX | DisableModes.DisableGyroscopeY | DisableModes.DisableGyroscopeZ));
+            // ACCEL_CONFIG 2 (0x1D) set ACCEL_FCHOICE_B = 0 and A_DLPFCFG[2:0]=1(b001)
+            // Bandwidth for Accelerator to 184Hz
+            AccelerometerBandwidth = AccelerometerBandwidth.Bandwidth0184Hz;
+            // Enable Motion Interrupt
+            //  In INT_ENABLE (0x38), set the whole register to 0x40 to enable motion interrupt only
+            WriteRegister(Register.INT_ENABLE, 0x40);
+            // Enable AccelHardware Intelligence:
+            // In MOT_DETECT_CTRL (0x69), set ACCEL_INTEL_EN = 1 and ACCEL_INTEL_MODE  = 1
+            WriteRegister(Register.MOT_DETECT_CTRL, 0b1100_0000);
+            // Set Motion Threshold:
+            // In WOM_THR (0x1F), set the WOM_Threshold[7:0] to 1~255 LSBs (0~1020mg)
+            WriteRegister(Register.WOM_THR, (byte)magnetometerThreshold);
+            // Set Frequency of Wake-up:
+            // In LP_ACCEL_ODR (0x1E), set Lposc_clksel[3:0] = 0.24Hz ~ 500Hz
+            WriteRegister(Register.LP_ACCEL_ODR, (byte)acceleratorLowPower);
+            // Enable Cycle Mode (AccelLow Power Mode):
+            // In PWR_MGMT_1 (0x6B) make CYCLE =1
+            WriteRegister(Register.PWR_MGMT_1, 0b0010_0000);
+            // Motion Interrupt Configuration Completed
+        }
+
+        /// <summary>
+        /// Initialize the MPU9250
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device</param>
+        /// <param name="autoDispose">Will automatically dispose the I2C device if true</param>
+        public Mpu9250(I2cDevice i2cDevice, bool autoDispose = true)
```

Comment:
We haven't been using autoDispose for most devices unless we have a ver obvious scenario where you wouldn't want to autoDispose (Like for example in the case of your Ak8963 where you do might not necesarilly want to autoDispose). So unless you have a clear scenario why not to autodispose, let's just remove this and always dispose the i2c device.